exporter/googlecloudstorageexporter/exporter.go

```diff
@@ -0,0 +1,136 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlecloudstorageexporter // import "github.com/observiq/bindplane-otel-collector/exporter/googlecloudstorageexporter"
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+// googleCloudStorageExporter exports OTLP data as Google Cloud Storage objects
+type googleCloudStorageExporter struct {
+	cfg           *Config
+	storageClient storageClient
+	logger        *zap.Logger
+	marshaler     marshaler
+}
+
+// newExporter creates a new Google Cloud Storage exporter
+func newExporter(cfg *Config, params exporter.Settings) (*googleCloudStorageExporter, error) {
+	storageClient, err := newGoogleCloudStorageClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create storage client: %w", err)
+	}
```

Comment:
General comment to apply through out the pr, but for errors can we make sure they're all wrapped with a message like is done here. However, we shouldn't include "failed to..." or "error while..." just because it'll be propagated as an error and when you get several errors deep it becomes a bit excessive.
```suggestion
	if err != nil {
		return nil, fmt.Errorf("create storage client: %w", err)
	}
```